packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -447,6 +511,8 @@ export class SnapInterfaceController extends BaseController<
       this.#checkPhishingList.bind(this),
       (id: string) => this.messagingSystem.call('SnapController:get', id),
     );
+
+    validateAssetSelector(element, this.#getAccountByAddress.bind(this));
```

Comment:
I think we need to refactor `#validateContent` for performance. Right now we would have to walk the JSX twice: once for links and once for the asset selector. This will only grow.

I'm thinking we should walk it once and have functions we can call to validate specific elements (instead of having `validateJsxLinks` etc handle the walking themselves).